herbert_core/classes/MPIFramework/@parallel_config/private/check_and_set_worker_.m

```diff
@@ -32,6 +40,23 @@
         
     end
 else % worker function is available.
-    config_store.instance().store_config(obj,'worker',new_wrkr);
+    config_store.instance().store_config(obj, 'worker', new_wrkr);
+    config_store.instance().store_config(obj, 'is_compiled', false);
 end
+end % function
 
+function out = check_compiled_(worker)
+    out = '';
+    if exist(worker, 'file')
+        % Assume if input is full path to file, then it is a compiled worker
+        out = worker;
+    else
+        if ispc(), cmd = 'where'; else, cmd = 'which'; end
+        [rs, rv] = system([cmd ' ' worker]);
+        if rs == 0
+            % Assume if it is on the system path, then it is a compiled worker
+            out = splitlines(strip(rv));
+            out = out{1}; % Only take first path if there is more than one
+        end
```

Comment:
So this bit is to allow a user shortcut because what we want is the fully qualified path to pass to the Java `ProcessBuilder`, so if the user just sets `pc.worker = 'horace_worker.exe'` without the full path it would not work; so the `system('[which|where] <worker>')` here is to get the full path. `.m` files will not be returned by this (unless they have erroneously been marked as executable by the filesystem on Linux).